flink-clients/src/main/java/org/apache/flink/client/program/StreamContextEnvironment.java

```diff
@@ -169,12 +236,36 @@ public static void setAsContext(
                             mergedConfiguration,
                             userCodeClassLoader,
                             enforceSingleJobExecution,
-                            suppressSysout);
+                            suppressSysout,
+                            allowConfigurations,
+                            errors);
                 };
         initializeContextEnvironment(factory);
     }
 
     public static void unsetAsContext() {
         resetContextEnvironment();
     }
+
+    private List<ConfigurationNotAllowedError> checkNotAllowedConfigurations() {
+        final List<ConfigurationNotAllowedError> errors = new ArrayList<>();
+        if (allowConfigurations) {
+            return errors;
+        }
+        final MapDifference<String, String> diff =
```

Comment:
When is `originalConfiguration` not empty?

Reply:
During the call of the parent ctor the `StreamExecutionEnvironment` translates the configuration into the equivalent sub configurations. [1]

[1] https://github.com/apache/flink/blob/7641c239b777273126c10efab334e3247771a28b/flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java#L277